src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTests.cs

```
@@ -0,0 +1,312 @@
+        }
+
+        private async Task<IAsyncDisposable> PushGitBranchAsync(string remote, string branch)
+        {
+            await RunGitAsync("push", remote, branch);
+            return AsyncDisposable.Create(async () =>
+            {
+                _output.WriteLine($"Cleaning up Remote branch {branch}");
+                await RunGitAsync("push", remote, "--delete", branch);
+            });
+        }
+
+        private string GetRepoUrl(string org, string repository)
+        {
+            return $"https://github.com/{org}/{repository}";
+        }
+
+        private string GetRepoUrl(string repository)
+        {
+            return GetRepoUrl(_parameters.GitHubTestOrg, repository);
+        }
+
+        private string GetRepoFetchUrl(string repository)
+        {
+            return GetRepoFetchUrl(_parameters.GitHubTestOrg, repository);
+        }
+
+        private string GetRepoFetchUrl(string org, string repository)
+        {
+            return $"https://{_parameters.GitHubUser}:{_parameters.GitHubToken}@github.com/{org}/{repository}";
+        }
+
+        private Task<string> RunDarcAsync(params string[] args)
+        {
+            return TestHelpers.RunExecutableAsync(_output, _parameters.DarcExePath, args.Concat(new[]
+            {
+                "-p", _parameters.MaestroToken,
+                "--bar-uri", _parameters.MaestroBaseUri,
+                "--github-pat", _parameters.GitHubToken,
+            }).ToArray());
+        }
+
+        private Task<string> RunGitAsync(params string[] args)
+        {
+            return TestHelpers.RunExecutableAsync(_output, _parameters.GitExePath, args);
+        }
+
+        private async Task<AsyncDisposableValue<string>> CreateTestChannelAsync(string testChannelName)
+        {
+            try
+            {
+                await RunDarcAsync("delete-channel", "--name", testChannelName).ConfigureAwait(false);
+            }
+            catch (XunitException) { }
```

... swallowing xunit exceptions???

---

This is catching the exception that is thrown when the called executable returns non 0. This is just a quick pre-cleanup so we don't fail creating the channel.